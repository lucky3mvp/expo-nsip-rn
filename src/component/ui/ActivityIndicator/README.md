---
title: ActivityIndicator
subTitle: 活动指示器 
---

# 活动指示器

### 组件描述
- 活动指示器

### 示例代码

```html
<ActivityIndicator />
<ActivityIndicator color="#33F" text="正在加载" />
<ActivityIndicator size="large" />
<ActivityIndicator text="正在加载" />
<ActivityIndicator size="large" color="#FFF"/>
<ActivityIndicator size="large" color="#F22" text="正在加载" />
```

## API

### props列表

属性 | 说明 | 类型 | 默认值
----|-----|------|------
| visible | 是否显示指示器 | boolean | true |
| color | 指示器颜色 | string | #999 |
| size | 指示器大小 | string | small |
| text | 指示器显示文本内容 | string | - |

### 样式对象styles

属性 | 说明 | 适用类型
----|-----|------
| container | 容器样式 | View |
| text | 文本样式 | Text |