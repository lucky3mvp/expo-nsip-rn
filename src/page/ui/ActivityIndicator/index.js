import React, { Component } from 'react'
import { ScrollView, View, Text } from 'react-native'
import { Style } from 'common'
import { ActivityIndicator, Button } from 'component'
export default class Page extends Component {
	static navigationOptions = {
		title: '活动指示器'
	}
	constructor(props){
		super(props)
		this.state = {
			toast1Visible: false,
			toast2Visible: false
		}
	}
	render() {
		return (
			<ScrollView style={Style.container}>
				<ActivityIndicator />
				<ActivityIndicator color="#33F" textStyle={{color: '#33F'}} text="正在加载" />
				<ActivityIndicator size="large" />
				<ActivityIndicator text="正在加载" />
				<ActivityIndicator visible={this.state.toast1Visible} hasToast={true} size="large" color="#FFF"/>
			</ScrollView>
		);
	}
}