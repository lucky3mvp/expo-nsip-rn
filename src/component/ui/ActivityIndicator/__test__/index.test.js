import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import ActivityIndicator from '../index'

describe('ActivityIndicator Tests', () => {
	it('renders correctly', () => {
		const tree = Renderer.create((
			<ActivityIndicator />
		)).toJSON();
		
		expect(tree).toMatchSnapshot();
	});

	it('text props', () => {
		let defaultWrapper = shallow(<ActivityIndicator />);
		expect(defaultWrapper.instance().props.text).toBeUndefined();

		let textWrapper = shallow(<ActivityIndicator text='加载呢，别急' />);
		expect(textWrapper.instance().props.text).toBe('加载呢，别急');
	});

	it('size props', () => {
		let defaultWrapper = shallow(<ActivityIndicator />);
		expect(defaultWrapper.instance().props.size).toBe('small');

		let largeWrapper = shallow(<ActivityIndicator size='large' />);
		expect(largeWrapper.instance().props.size).toBe('large');
	});
})
