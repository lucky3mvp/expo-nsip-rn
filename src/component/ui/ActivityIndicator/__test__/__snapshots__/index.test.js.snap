// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator Tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
